onnxruntime/contrib_ops/rocm/bert/multihead_attention.cu

```diff
@@ -52,11 +85,30 @@ Status MultiHeadAttention<T>::ComputeInternal(OpKernelContext* context) const {
   const Tensor* query = context->Input<Tensor>(0);
   const Tensor* key = context->Input<Tensor>(1);
   const Tensor* value = context->Input<Tensor>(2);
-  const Tensor* bias = context->Input<Tensor>(3);
-  const Tensor* key_padding_mask = context->Input<Tensor>(4);
-  const Tensor* relative_position_bias = context->Input<Tensor>(5);
-  const Tensor* past_key = context->Input<Tensor>(6);
-  const Tensor* past_value = context->Input<Tensor>(7);
+
+  const Tensor* bias{};
+  const Tensor* key_padding_mask{};
+  const Tensor* relative_position_bias{};
+  const Tensor* past_key{};
+  const Tensor* past_value{};
+  const Tensor* past_seq_len{};
+
+  if (attn_type_ == MHA) {
+    bias = context->Input<Tensor>(3);
+    key_padding_mask = context->Input<Tensor>(4);
+    relative_position_bias = context->Input<Tensor>(5);
+    past_key = context->Input<Tensor>(6);
+    past_value = context->Input<Tensor>(7);
+  } else if (attn_type_ == DMMHA) {
+    key_padding_mask = context->Input<Tensor>(3);
+    relative_position_bias = context->Input<Tensor>(4);
+    past_key = context->Input<Tensor>(5);
+    past_value = context->Input<Tensor>(6);
+    past_seq_len = context->Input<Tensor>(kPastSequenceLengthInputIndex);
+    // const Tensor* beam_width = context->Input<Tensor>(8);         // NOTE: not used
+    // const Tensor* cache_indirection = context->Input<Tensor>(9);  // TODO: should not present for ROCm EP
```

Comment:
How about add a check ORT_ENFORCE(cache_indirection==nullptr)?

Reply:
Currently, it is not easy to do so. They are coupled deeply with decoding logic and they are passed in unconditionly. So these lines are only commented out, instead of ORT_ENFORCE.